operate/webapp/src/main/java/io/camunda/operate/webapp/backup/BackupRepositoryWrapper.java

```diff
@@ -0,0 +1,104 @@
+import io.camunda.operate.exceptions.OperateOpensearchConnectionException;
+import io.camunda.operate.exceptions.OperateRuntimeException;
+import io.camunda.operate.webapp.elasticsearch.backup.ElasticsearchBackupRepository;
+import io.camunda.operate.webapp.opensearch.backup.OpensearchBackupRepository;
+import io.camunda.webapps.backup.BackupRepository;
+import io.camunda.webapps.backup.BackupService.SnapshotRequest;
+import io.camunda.webapps.backup.GetBackupStateResponseDto;
+import io.camunda.webapps.backup.exceptions.InvalidRequestException;
+import io.camunda.webapps.backup.exceptions.ResourceNotFoundException;
+import io.camunda.webapps.backup.repository.BackupRepositoryConnectionException;
+import io.camunda.webapps.backup.repository.GenericBackupException;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Wraps a BackupRepository in order to map the exception thrown by the generic implementation to
+ * the exception that operate expects to be raised:
+ *
+ * <ul>
+ *   <li>{@link io.camunda.operate.exceptions.OperateElasticsearchConnectionException}
+ *   <li>{@link io.camunda.operate.exceptions.OperateOpensearchConnectionException}
+ *   <li>{@link io.camunda.operate.exceptions.OperateRuntimeException}
+ *   <li>{@link io.camunda.operate.webapp.api.v1.exceptions.ResourceNotFoundException}
+ * </ul>
+ */
+public class BackupRepositoryWrapper implements BackupRepository {
+
+  private final BackupRepository repository;
+
+  BackupRepositoryWrapper(final BackupRepository repository) {
+    this.repository = repository;
+  }
+
+  @Override
+  public void deleteSnapshot(final String repositoryName, final String snapshotName) {
+    remap(() -> repository.deleteSnapshot(repositoryName, snapshotName));
+  }
+
+  @Override
+  public void validateRepositoryExists(final String repositoryName) {
+    remap(() -> repository.validateRepositoryExists(repositoryName));
+  }
+
+  @Override
+  public void validateNoDuplicateBackupId(final String repositoryName, final Long backupId) {
+    remap(() -> repository.validateNoDuplicateBackupId(repositoryName, backupId));
+  }
+
+  @Override
+  public GetBackupStateResponseDto getBackupState(
+      final String repositoryName, final Long backupId) {
+    return remap(() -> repository.getBackupState(repositoryName, backupId));
+  }
+
+  @Override
+  public List<GetBackupStateResponseDto> getBackups(final String repositoryName) {
+    return remap(() -> repository.getBackups(repositoryName));
+  }
+
+  @Override
+  public void executeSnapshotting(
+      final SnapshotRequest snapshotRequest, final Runnable onSuccess, final Runnable onFailure) {
+    remap(() -> repository.executeSnapshotting(snapshotRequest, onSuccess, onFailure));
+  }
+
+  private void remap(final Runnable runnable) {
+    remap(
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  private <A> A remap(final Supplier<A> supplier) {
```

Comment:
Remapping of the "generic exceptions" from webapp-backups to the ones that are expected by the controller